### PR TITLE
Proposal: POMP

### PR DESCRIPTION
**Pull request by zhangjing28**
#### Project Abstract
To prove ownership of DOT or other Polkadot Parachain cryptocurrencies, we need to prove that we know the private key. However, we would not want to reveal the private key, as revealing even part of the private key or an image of the private key could put our ownership at risk. By having the private key — or even part of it and then using sophisticated software to attempt to guess the rest of it — hackers can steal your cryptocurrency. 

The most reliable way to prove ownership of cryptocurrencies is to sign a specified message with your private key. By doing so, you can prove ownership of your crypto assets without the need to reveal the key or having to engage in a transaction, even a small amount transfer.

The POMP platform enables all crypto token owners of Polkadot Parachain to generate their encrypted proof key through zero knowledge and prove crypto assets without the risk that may be leaking the private key.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [X] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [X] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [X] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [X] The software delivered for this grant will be released under an open-source license specified in the application.
- [X] The initial PR contains only one commit (squash and force-push if needed).
- [X] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by keeganquigley** [comment]
Thanks for the proposal @zhangjing28 a couple comments:

- Can you explain the technical details regarding the "PC website"?  Would this be a front-end and what framework/language would be used?
- We usually ask for mockups to be provided in the application for front-end components. 
- Usually for teams we haven't worked with before, we recommend going for a level 1 or 2 PoC to give us a chance to see some of your work, followed by subsequent grants. Would you be willing to perhaps reduce the scope?

**Comment by zhangjing28** [comment]
> Thanks for the proposal @zhangjing28 a couple comments:
> 
> * Can you explain the technical details regarding the "PC website"?  Would this be a front-end and what framework/language would be used?
> * We usually ask for mockups to be provided in the application for front-end components.

Hi, @keeganquigley , thanks for the reply.

* The PC website which means we will deliver a front-end website in the milestone. It will be developed by React.js and Typescript.
* I already added our UI design mockup pictures into the application.

File: applications/POMP.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# POMP (Prove of My Possession)
```
Comment (keeganquigley) [inline applications/POMP.md:1]:
```suggestion
# POMP (Proof of My Possession)
```

File: applications/POMP.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 2
+
+### Overview
+
+Cryptocurrencies are digital coins and tokens that represent real-world value without relying on a central issuer. Instead of going through a bank or a government, cryptocurrencies rely on cryptography, public ledgers, and blockchain technology to verify ownership. It’s playing a more and more important role in the real world. Let's just look at the numbers. Bitcoin had a market capitalization of $510 Billion on June 8, 2023. On the same day, Ethereum had a market capitalization of $220 Billion.
+
+However, it’s not easy to prove that you are the owner of one specific blockchain account. Because you have to import a private key to a wallet and sign a signature to prove that the private key is matching to the account’s public key. Additionally, the more a private key is imported, the more it opens up the threat vector for either intentional or unintentional damage.
+
+Many individuals and companies prefer to have an additional layer of asset proof to keep their privacy. That’s where the POMP(Prove of My Possession) comes to play.
```
Comment (keeganquigley) [inline applications/POMP.md:13]:
```suggestion
Many individuals and companies prefer to have an additional layer of asset proof to keep their privacy. That’s where the POMP(Proof of My Possession) comes to play.
```

File: applications/POMP.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+### Project Details
+
+The POMP is a decentralized service that combines zero-knowledge technology, users can easily generate encrypted proof keys while maintaining the privacy of their crypto assets. After generated, the user can prove the ownership of the crypto asset to any third party without inputting any private key.
+
+Here is the basic workflow for generating:
```
Comment (keeganquigley) [inline applications/POMP.md:19]: So if I understand correctly, the user has to have Manta wallet in order to use the tool, correct?

File: applications/POMP.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+- **FTE:**  3
+- **Costs:** 30,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Apache 2.0 |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user sends test transactions, which will show how the new platform works. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Article | We will publish an **article** that explains what this grant achieved.|
+| 1. | Javascript Library | We will deliver a JS library that will implement the functionality described under the "Workflow" part, including recovering the address from the signature, verifying the asset amount, hashing the ZK proof key, and integrating it with the ink!  smart contract functions.|
```
Comment (keeganquigley) [inline applications/POMP.md:88]: Will other devs be able to use this library to integrate the tool into their dApp? Or does the user essentially have to "opt in" by using Manta wallet to generate the proof?

File: applications/POMP.md (hunk 3 above)
Comment (zhangjing28) [reply applications/POMP.md:19]: Yes, the user has to have a Manta wallet to generate the zero-knowledge data.

File: applications/POMP.md (hunk 4 above)
Comment (zhangjing28) [reply applications/POMP.md:88]: Sure, any other devs are able to integrate our JS library to verify the proof key that is saved on ink! smart contract. And we have to use the Manta wallet to generate the zero-knowledge data when the user would like to create Proof of Possession. Which means generate once, verify anywhere.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. I have one initial question: What is the advantage of your solution compared to using, for example, [remarks](https://wiki.polkadot.network/docs/glossary#remarks) or simply signing something with your private key off-chain?

**Comment by zhangjing28** [comment]
> Thanks for the application. I have one initial question: What is the advantage of your solution compared to using, for example, [remarks](https://wiki.polkadot.network/docs/glossary#remarks) or simply signing something with your private key off-chain?

Hi, @Noc2 , thanks for your reply. 

For most users, the Remarks or signing something with their private key off-chain is not easy to implement. They have to read the developer documentation and install the dev environment. It's a good way for users with a technology background, but not for regular users.

The POMP is trying to enable all Polkadot users to prove their crypto assets by signing the specific message with Polkadot.js & Manta wallet. In this way, the learning cost and use costs of all users are greatly reduced. 

Additionally. The POMP is an asset-proving platform not just a tool. The users can prove that they are the owners of some crypto asset (token or NFT) by just inputting the proof key into any other third products, such as SocialFi. It's like an asset proof issued by the bank, so we can use it to prove how much money we have.

**Comment by keeganquigley** [comment]
Thanks for the changes and replies @zhangjing28 I will go ahead and mark the application as ready for review and ping the rest of the committee so that others can comment.

Personally, I won't approve it yet as I feel that the price is a bit high. I'd love to see it fit into a level 2 PoC of the smart contracts and library, perhaps with a follow-up grant for the front-end. But perhaps others will feel differently, so feel free to argue your position.

**Comment by zhangjing28** [comment]
> Thanks for the changes and replies @zhangjing28 I will go ahead and mark the application as ready for review and ping the rest of the committee so that others can comment. It's an interesting concept for the abstraction of a seed phrase.
> 
> Personally, I won't approve it yet as I feel that the price is a bit high. I'd love to see it fit into a level 2 PoC of the smart contracts and library, perhaps with a follow-up grant for the front-end. But perhaps others will feel differently, so feel free to argue your position.

Thanks for your valuable feedback @keeganquigley . I already changed the total cost from 30,000 to 22,000 USD.

The front end is an important part of the project as a whole. We still would like to put it into the deliveries. We will cover the development fee for it ourselves.

**Review by takahser — CHANGES_REQUESTED** [review]
@zhangjing28 I think it would be helpful for understanding the full potential of POMP if you could outline some specific use cases or scenarios where you envision POMP being utilized. Could you share any potential applications, industries, or situations where you see POMP being particularly useful?

**Comment by zhangjing28** [comment]
> @zhangjing28 I think it would be helpful for understanding the full potential of POMP if you could outline some specific use cases or scenarios where you envision POMP being utilized. Could you share any potential applications, industries, or situations where you see POMP being particularly useful?

Hi, @takahser Thanks for your reply.

One of the most interesting scenarios that we are working on is the SocialFi Application. Let's say I'm a user of the SocialFi App, I would like to prove I am the owner of a specific account with 100,0000 DOT. Or I'm the owner of the specific NFT collection. Because that will increase my attraction to other users. But I don't want to import my private key to any third App, Polkadot.js wallet is the only one that I trust. So that's where the POMP comes to play.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the updates here. To be honest, I'm not fully convinced by your use cases and examples here. I still think it's a lot easier to simply sign a transaction or remarks. For usability, ideally, wallets integrate something for this. Using a separate UI specifically for this purpose makes it not really user-friendly. So I'm personally not willing to support it, but others might have a different opinion here.

**Comment by zhangjing28** [comment]
> Thanks for the updates here. To be honest, I'm not fully convinced by your use cases and examples here. I still think it's a lot easier to simply sign a transaction or remarks. For usability, ideally, wallets integrate something for this. Using a separate UI specifically for this purpose makes it not really user-friendly. So I'm personally not willing to support it, but others might have a different opinion here.

@Noc2  Hi, thanks for your valuable reply.

The POMP is not just a tool for Polkadot.js wallet, but the first online private asset verifier to prove your assets without revealing addresses, ensuring privacy and data protection for the whole blockchain ecosystem. We will support Bitcoin/Ethereum/Binance Chain and other blockchains in the future. So everyone could use their wallet to sign a specific message to prove their ownership of any blockchain without the wallet upgrade.

It's like a crypt asset protocol for the blockchain ecosystem. There are 3 roles here.

* The users: Token holders who would like to prove their on-chain asset without revealing details.
* The Dapps: Web3 Dapps would like to allow the users to prove their on-chain assets.
* The POMP: The private asset verifier to generate proof-key for users.

We think this is a good business model that solves the situation in which users want assets on the chain of privacy certificates.

**Comment by semuelle** [comment]
Since the application hasn't garnered any approvals, I am going to close it. Thanks for the application and engaging with us, @zhangjing28. 

Please feel free to apply again in the future.

**Pull request closed without merging**